blockchain/blockdao/blockindexer.go

```diff
@@ -74,14 +67,7 @@ func (bic *BlockIndexerChecker) CheckIndexer(ctx context.Context, indexer BlockI
 	if targetHeight == 0 || targetHeight > daoTip {
 		targetHeight = daoTip
 	}
-	startHeight := tipHeight + 1
-	if indexerWS, ok := indexer.(BlockIndexerWithStart); ok {
-		indexStartHeight := indexerWS.StartHeight()
-		if indexStartHeight > startHeight {
-			startHeight = indexStartHeight
-		}
-	}
-	for i := startHeight; i <= targetHeight; i++ {
+	for i := tipHeight + 1; i <= targetHeight; i++ {
```

Comment:
currently `tipHeight` at L52 returns 0, so we need to extra code to check every indexer's start height, which is somewhat confusing, since the same code (to check every indexer's start height) also appears in `SyncIndexer`. So that should be handles by `SyncIndexer`, and we only need to get it's height (at L52)